src/systems/systemDialog.component.tsx

```diff
@@ -27,185 +27,159 @@ import {
-import { trimStringValues } from '../utils';
-
-export type SystemDialogType = 'add' | 'edit' | 'save as';
-
-const getEmptySystem = (): SystemPost => {
-  return {
-    // Here using null for optional values only, so that types for isUpdated parameters
-    // can match
-    name: '',
-    description: null,
-    location: null,
-    owner: null,
-    importance: SystemImportanceType.MEDIUM,
-  } as SystemPost;
-};
 
 export interface SystemDialogProps {
   open: boolean;
   onClose: () => void;
-  type: SystemDialogType;
+  requestType: RequestType;
+  saveAs?: boolean;
   // Only required for add
   parentId?: string | null;
   // Only required for pre-populating fields for an edit dialog
   selectedSystem?: System;
 }
 
 const SystemDialog = React.memo((props: SystemDialogProps) => {
-  const { open, onClose, parentId, type, selectedSystem } = props;
+  const { open, onClose, parentId, requestType, selectedSystem, saveAs } =
+    props;
 
-  // User entered properties
-  const [systemData, setSystemData] =
-    React.useState<SystemPost>(getEmptySystem());
+  const isNotCreating = (requestType !== 'post' || saveAs) && selectedSystem;
 
-  // Ensure system data is updated when the selected system changes
-  useEffect(() => {
-    if (open) {
-      if (type === 'add') setSystemData(getEmptySystem());
-      else if (selectedSystem) setSystemData(selectedSystem as SystemPost);
-    }
-  }, [selectedSystem, open, type]);
+  const { mutateAsync: postSystem, isPending: isAddPending } = usePostSystem();
+  const { mutateAsync: patchSystem, isPending: isEditPending } =
+    usePatchSystem();
 
-  // Error messages for the above properties (undefined means no error)
-  const [nameError, setNameError] = React.useState<string | undefined>(
-    undefined
+  const initialSystem: SystemPost = React.useMemo(
+    () =>
+      isNotCreating
+        ? selectedSystem
+        : {
+            name: '',
+            description: '',
+            location: '',
+            owner: '',
+            importance: SystemImportanceType.MEDIUM,
+          },
+    [isNotCreating, selectedSystem]
   );
 
-  // Form error that should disappear when the form is modified
-  const [formError, setFormError] = React.useState<string | undefined>(
-    undefined
-  );
+  const {
+    handleSubmit,
+    register,
+    formState: { errors },
+    watch,
+    control,
+    setError,
+    clearErrors,
+    reset,
+  } = useForm<SystemPost>({
+    resolver: zodResolver(SystemsSchema(requestType)),
+    defaultValues: initialSystem,
+  });
 
-  const handleClose = React.useCallback(() => {
-    if (type === 'add') setSystemData(getEmptySystem());
-    // Reset for edit
-    else setSystemData(selectedSystem as SystemPost);
+  // Load the values for editing.
+  React.useEffect(() => {
+    reset(initialSystem);
+  }, [initialSystem, reset]);
 
-    // Remove all errors
-    setNameError(undefined);
-    setFormError(undefined);
+  React.useEffect(() => {
```

Comment:
I think it would be nice for this and the one in manufacturers to have  a comment on when its clearing the errors.